src/test/java/com/automatedtest/sample/definitions/SearchSettingPageSteps.java

```diff
@@ -0,0 +1,58 @@
+package com.automatedtest.sample.definitions;
+
+import com.automatedtest.sample.driver.Wait;
+import com.automatedtest.sample.pages.HomePage;
+import com.automatedtest.sample.pages.SettingPage;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.junit.Assert;
+
+public class SearchSettingPageSteps {
+
+    private final SettingPage settingPage;
+
+    public SearchSettingPageSteps() {
+        this.settingPage = new SettingPage();
+    }
+
+    @Given("I'm on Google Search Setting Page")
+    public void aUserNavigatesToSearchSettingPage() {
+        this.settingPage.goToSettingPage();
+    }
+
+    @When("I Check Turn {string} SafeSearch")
+    public void aUserCheckTurnOnOffSafeSearch(String onOff) {
+        if(!onOff.equalsIgnoreCase(this.settingPage.getSafeSearchValue()))
+            this.settingPage.clickChangeSafeSearch();
+    }
+
+    @And("I save the Setting Page")
+    public void aUserSaveTheSettingPage() {
+        this.settingPage.saveSettingPage();
+        this.settingPage.acceptConfirmDialog();
+        this.settingPage.delay(1);
+    }
+
+    @And("I comeback Setting Page again")
+    public void aUserComebackSettingPageAgain() {
+        this.settingPage.goToSettingPage();
```

Comment:
we should use the UI to navigate to Setting Page, we should use URL only at Given step